rpcserver.go

```diff
@@ -3420,25 +3420,85 @@ func (r *rpcServer) GetNodeInfo(ctx context.Context,
 func (r *rpcServer) QueryRoutes(ctx context.Context,
 	in *lnrpc.QueryRoutesRequest) (*lnrpc.QueryRoutesResponse, error) {
 
-	// First parse the hex-encoded public key into a full public key object
-	// we can properly manipulate.
-	pubKeyBytes, err := hex.DecodeString(in.PubKey)
-	if err != nil {
-		return nil, err
+	// If a payment request was specified, extract the destination, amount,
+	// and hops from it.
+	payReq, err := zpay32.Decode(
+		in.PaymentRequest, activeNetParams.Params,
+	)
+	if err != nil && len(in.PaymentRequest) > 0 {
```

Comment:
should check if `PaymentRequest` is present _before_ attempting to parse it.

Reply:
Yea. I was taking advantage of Decode handling correctly null inputs. Anyhow, I agree that this is convoluted. Rearranged code to make it more readable.